src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1280,7 +1281,69 @@ static void fill_idPtypeNP
             "return false; " +
          "}";
   LM_T(LmtMongo, ("idTtypeNP function: %s", bgP->functionIdPtypeNP.c_str()));
-
+#else
+  orion::BSONObjBuilder outer_obj;
+  orion::BSONObjBuilder anyElementTrue;
+  orion::BSONObjBuilder map_obj;
+  orion::BSONObjBuilder in;
+  orion::BSONArrayBuilder and_arr;
+  orion::BSONObjBuilder and_first;
+  orion::BSONObjBuilder and_second;
+  orion::BSONObjBuilder and_third;
+  orion::BSONObjBuilder and_fourth;
+  orion::BSONObjBuilder regex_obj;
+  orion::BSONArrayBuilder eq_arr;
+  orion::BSONObjBuilder eq_obj;
+  orion::BSONObjBuilder eq_obj_2;
+  orion::BSONObjBuilder eq_obj_3;
+  orion::BSONObjBuilder type_obj;
+  orion::BSONArrayBuilder eq_arr_2;
+  orion::BSONArrayBuilder eq_arr_3;
+  orion::BSONArrayBuilder eq_arr_4;
+  orion::BSONArrayBuilder eq_arr_5;
+  orion::BSONArrayBuilder or_arr;
+
+  map_obj.append("input", "$entities");
+  
+  regex_obj.append("input", entityId);
+  regex_obj.append("regex", "$$this.id");
+  regex_obj.append("options", "i");
+  and_first.append("$regexMatch", regex_obj.obj());
+  and_arr.append(and_first.obj());
+  
+  eq_arr.append("$$this.isPattern");
+  eq_arr.append("true");
+  and_second.append("$eq", eq_arr.arr());
+  and_arr.append(and_second.obj());
+
+  eq_arr_2.append("$$this.type");
+  eq_arr_2.append(entityType);
+  eq_obj.append("$eq", eq_arr_2.arr());
+  or_arr.append(eq_obj.obj());
+
+  eq_arr_3.append("$$this.type");
+  eq_arr_3.append("");
+  eq_obj_2.append("$eq", eq_arr_3.arr());
+  or_arr.append(eq_obj_2.obj());
+  
+  eq_arr_5.append("$$this.type");
+  eq_arr_5.append(NULL);
+  eq_obj_3.append("$eq", eq_arr_5.arr());
+  or_arr.append(eq_obj_3.obj());
+  
+  and_third.append("$or", or_arr.arr());
+  and_arr.append(and_third.obj());
+
+  type_obj.append("$type", "$$this.isTypePattern");
+  eq_arr_4.append(type_obj.obj());
+  eq_arr_4.append(false);
```

Comment:
Not sure if I'm missing something... but it seems that eq_arr_4 is never appended to the holding element...